From e8e8e05538bf2c7bd8feebc44d4a960f453d21e1 Mon Sep 17 00:00:00 2001
From: holyfei <yangfeiyu20092010@163.com>
Date: Sat, 20 Nov 2021 17:04:39 +0800
Subject: [PATCH] kata-containers: modify kernel and image path in
 configuration.toml

Signed-off-by: holyfei <yangfeiyu20092010@163.com>
---
 src/runtime/Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/runtime/Makefile b/src/runtime/Makefile
index 745bcc1..68e60ee 100644
--- a/src/runtime/Makefile
+++ b/src/runtime/Makefile
@@ -49,6 +49,8 @@ BINLIBEXECLIST :=
 BIN_PREFIX = $(PROJECT_TYPE)
 PROJECT_DIR = $(PROJECT_TAG)
 IMAGENAME = $(PROJECT_TAG).img
+KERNEL_PATH = /var/lib/kata/kernel
+IMAGE_PATH = /var/lib/kata/kata-containers-initrd.img
 
 TARGET = $(BIN_PREFIX)-runtime
 TARGET_OUTPUT = $(CURDIR)/$(TARGET)
@@ -108,7 +110,7 @@ PKGLIBEXECDIR := $(LIBEXECDIR)/$(PROJECT_DIR)
 
 KERNELDIR := $(PKGDATADIR)
 
-IMAGEPATH := $(PKGDATADIR)/$(IMAGENAME)
+IMAGEPATH := $(IMAGE_PATH)
 FIRMWAREPATH :=
 
 # Name of default configuration file the runtime will use.
@@ -248,8 +250,7 @@ ifneq (,$(QEMUCMD))
     DEFBLOCKSTORAGEDRIVER_QEMU := virtio-scsi
     DEFNETWORKMODEL_QEMU := tcfilter
     KERNELTYPE = uncompressed
-    KERNELNAME = $(call MAKE_KERNEL_NAME,$(KERNELTYPE))
-    KERNELPATH = $(KERNELDIR)/$(KERNELNAME)
+    KERNELPATH = $(KERNEL_PATH)
 endif
 
 ifneq (,$(CLHCMD))
@@ -294,8 +295,7 @@ ifneq (,$(STRATOVIRTCMD))
 	DEFBLOCKSTORAGEDRIVER_STRATOVIRT := virtio-mmio
 	DEFNETWORKMODEL_STRATOVIRT := none
 	KENRELTYPE_STRATOVIRT = uncompressed
-	KERNEL_NAME_STRATOVIRT = $(call MAKE_KERNEL_NAME,$(KENRELTYPE_STRATOVIRT))
-	KERNELPATH_STRATOVIRT = $(KERNELDIR)/$(KERNEL_NAME_STRATOVIRT)
+	KERNELPATH_STRATOVIRT = $(KERNEL_PATH)
 endif
 
 ifneq (,$(FCCMD))
-- 
2.27.0

